FF13Rando/Randomizers/BattleRando.cs

```diff
@@ -304,14 +355,29 @@ public override void Randomize()
                         if (possible.Count > 0)
                         {
                             //Select a new random enemy from the list if we have any to save
-                            e.sEntryBtChSpec_string = RandomNum.SelectRandomWeighted(possible, _ => 1);
+                            e.sEntryBtChSpec_string = RandomNum.SelectRandom(possible);
                         }
                         else
                         {
                             Console.WriteLine($"Unable to resolve possible enemy shuffle for encounter {id}");
                         }
                     });
                 });
+
+                //Step 3: Remove any unused enemies from the character set to reduce memory overhead.
+                Randomizers.SetUIProgress("Cleaning up unused enemies from character sets", 3, 3);
+                charasets.ForEach(charaset =>
+                {
+                    //TODO: Filter this to be non-battle characters only?
+                    List<string> original = charaSetsOrig[charaset];
+                    //Extract all battles for a given charaset.
+                    List<string> battles = battleData.Where(battle => battle.Value.Charasets.Contains(charaset)).Select(b => b.Key).ToList();
+                    //Extract all used enemies from battles
+                    List<string> used = battles.SelectMany(b => btscs[b].Values.Select(e => e.sEntryBtChSpec_string)).Distinct().ToList();
```

Comment:
Found the issue regarding PSICOM enemies and others.

The sCharaSpec_string ID should be used here instead of e.sEntryBtChSpec_string:
`enemyRando.btCharaSpec[e.sEntryBtChSpec_string].sCharaSpec_string`

sCharaSpec_string is the model/charaset ID in the charasets WDB. 
sEntryBtChSpec_string is the actual enemy ID.

the most notable mismatched IDs are human enemies and Goblin.